autosubmitconfigparser/config/basicconfig.py

```diff
@@ -75,19 +76,19 @@ def props(self):
     def expid_dir(exp_id):
         if not isinstance(exp_id, str) or len(exp_id) != 4 or "/" in exp_id:
             raise TypeError("Experiment ID must be a string of 4 characters without the folder separator symbol")
-        return os.path.join(BasicConfig.LOCAL_ROOT_DIR, exp_id)
+        return Path(BasicConfig.LOCAL_ROOT_DIR).joinpath(exp_id)
 
     @staticmethod
     def expid_tmp_dir(exp_id):
-        return os.path.join(BasicConfig.expid_dir(exp_id), BasicConfig.LOCAL_TMP_DIR) 
+        return BasicConfig.expid_dir(exp_id).joinpath(BasicConfig.LOCAL_TMP_DIR) 
 
     @staticmethod
     def expid_log_dir(exp_id):
-        return os.path.join(BasicConfig.expid_tmp_dir(exp_id), f'LOG_{exp_id}')
+        return BasicConfig.expid_tmp_dir(exp_id).joinpath(f'LOG_{exp_id}')
 
     @staticmethod
     def expid_aslog_dir(exp_id):
-        return os.path.join(BasicConfig.expid_tmp_dir(exp_id), BasicConfig.LOCAL_ASLOG_DIR)
+        return BasicConfig.expid_tmp_dir(exp_id).joinpath(BasicConfig.LOCAL_ASLOG_DIR)
```

Comment:
In these examples I think either `joinpath` or `/` are OK (both call the same function, so whichever we prefer we probably should document somewhere so the code is uniform.

Reply:
I'm sticking with `joinpath` because of readibility (also discussed in [Autosubmit's MR](https://earth.bsc.es/gitlab/es/autosubmit/-/merge_requests/509)); pending documentation